### DISCUSSION TITLE
Tasks not using whole cpu

### DISCUSSION THREAD

**Discussion by ria8651** [bevyengine/bevy#3360] [Q&A] [answered]
I have 8 cores on my computer but [thread_num](https://docs.rs/bevy/0.5.0/bevy/tasks/struct.AsyncComputeTaskPool.html#method.thread_num) returns 2. When running tasks bevy only uses half my CPU. how can I use my whole CPU for async tasks?

Thanks

**Comment by bjorn3** [comment, marked as answer]
You need to insert a `DefaultTaskPoolOptions` resource into the `World` that specifies how much threads are dedicated to what kind of task (io, async compute, compute) By default half of all threads are assigned to compute: https://github.com/bevyengine/bevy/blob/cf221f9659127427c99d621b76c8085c4860e2ef/crates/bevy_core/src/task_pool_options.rs#L56-L81

**Reply by ria8651** [reply]
I can see the struct TaskPoolThreadAssignmentPolicy [here](https://github.com/bevyengine/bevy/blob/main/crates/bevy_core/src/task_pool_options.rs) but it doesn't seem to be public? It also doesn't show up on [docs.rs](https://docs.rs/bevy/0.5.0/bevy/?search=TaskPoolThreadAssignmentPolicy). How do I create an instance of this struct?

**Reply by bjorn3** [reply]
Looks like only `DefaultTaskPoolOptions` is exported. (https://github.com/bevyengine/bevy/blob/4c3c4b5e401717abbf57c56ef00a2e68ef7bce0d/crates/bevy_core/src/lib.rs#L12) Seems like an oversight. I think `DefaultTaskPoolOptions::default()` and then accessing the fields will work though.
